### ISSUE TITLE
Treat jobs with non-zero exit status as failures

### ISSUE THREAD

**Issue by cpick** [Canop/bacon#28]
Thanks for such a useful tool!

In addition to the standard jobs, I'd like to use `bacon` with some custom jobs.  When these jobs fail they exit with a non-zero status and print some useful error messages.

I propose changing `bacon`s behavior so that it does the following:
1. Try to parse and display warnings and errors as it normally does.  If it finds any, use current behavior (ie: display them in a structured fashion and report the warning/failure count).
2. Otherwise (no warnings and errors were detected in stderr), check the job process' exit status.  If it's 0, report "pass!" and display no output just as it currently does.
3. Otherwise (if exit status was non-zero), report failure and display the (unparsed) contents of stderr (as opposed to the current behavior of reporting "pass!" and displaying no output).

The idea is that this would support lots of new kinds of jobs right away and if new parsers are ever added, the output just gets better (and more structured) over time.

If this sounds like a reasonable feature I'd be happy to take a try at implementing it.

Thanks for considering the idea!

**Comment by Canop** [comment]
This looks reasonable. I think it's a good idea.

If you want to try coding it yourself, you may, as I don't have anything in progress in bacon, but be warned that I have a policy of not trying to merge code that I don't like and I'll ask you to not touch anything not related to your topic.

Can you tell me more about your use case (here or, better, in [the chat](https://miaou.dystroy.org/3768) during West European hours) ?

**Comment by Canop** [comment]
Noting here another case of failure which justifies this: a typo in `[patch.crates-io]`.

Bacon sees no problem while...

![image](https://user-images.githubusercontent.com/617006/102798898-090cb700-43b2-11eb-9efb-4f8551960f96.png)


TLDR: this becomes urgent
